src/widgets/list.rs

```diff
@@ -120,7 +113,12 @@ where
     }
 
     pub fn highlight_style(mut self, highlight_style: Style) -> List<'b, L> {
-        self.highlight_style = highlight_style;
+        self.highlight_style_diff = highlight_style.into();
+        self
+    }
+
+    pub fn highlight_style_diff(mut self, highlight_style: StyleDiff) -> List<'b, L> {
```

Comment:
I don't think that we need a second method here:

```rust
pub fn highlight_style<S>(mut self, highlight_style: S) -> List<'b, L> where S: Into<StyleDiff> {
}
```
The documentation should probably includes an example for both cases: `Style` and `StyleDiff` and explain how they differ. What do you think ?

Reply:
Actually reading your PR again, this might be confusing to some users. I think we will keep your new `diff` variants and deprecate the old ones.